### ISSUE TITLE
npm_import and lifecycle-hooks occasional fail due to package.json short-reads

### ISSUE THREAD

**Issue by jbedard** [aspect-build/rules_js#483]
For example:
```
SyntaxError: Unexpected end of JSON input
    at makeBins (/home/circleci/.cache/bazel/_bazel_circleci/.../bin/external/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:42097)
    at async main (/home/circleci/.cache/bazel/_bazel_circleci/.../bin/external/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:43024)
    at async /home/circleci/.cache/bazel/_bazel_circleci/.../bin/external/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:44054
```

**Comment by Kartones** [comment]
Also seeing the issue with v1.4.0, despite the retry introduced in that release:
```
Error parsing @swc/core-win32-arm64-msvc/package.json: SyntaxError: Unexpected end of JSON input

""""

""""
Retrying...
Error: Error parsing @swc/core-win32-arm64-msvc/package.json: SyntaxError: Unexpected end of JSON input

""""

""""
at makeBins ([redacted]/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:42314)
at async makeBins ([redacted]/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:41777)
at async main ([redacted]/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:43267)
at async [redacted]/aspect_rules_js/npm/private/lifecycle/lifecycle-hooks.sh.runfiles/aspect_rules_js/npm/private/lifecycle/min/index.min.js:73:44297
```

It doesn't looks as package dependant, seen it 3 times this week each with different packages.

**Comment by gregmagolan** [comment]
I've seen it locally on macos as well. What is strange is that it looks like in your logs above
```
const json = await fs.promises.readFile(packageJsonPath)
```
succeeds but returns an empty string. Twice in a row.

**Comment by gregmagolan** [comment]
I think I found the bug. Bad async/await logic in the lifecycle hooks. Will put up a PR to fix,

**Comment by gregmagolan** [comment]
Landed and included in v1.5.3 release
